ui/src/Queues/QueueRow.tsx

```diff
@@ -1,37 +1,87 @@
 import TableRow from "@material-ui/core/TableRow";
 import TableCell from "@material-ui/core/TableCell";
 import IconButton from "@material-ui/core/IconButton";
-import {KeyboardArrowDown, KeyboardArrowRight} from "@material-ui/icons";
-import React, {useState} from "react";
-import {QueueMessagesData} from "./QueueMessageData";
+import {
+  AddComment,
+  KeyboardArrowDown,
+  KeyboardArrowRight,
+} from "@material-ui/icons";
+import React, { useState } from "react";
+import { QueueMessagesData } from "./QueueMessageData";
 import RowDetails from "./QueueRowDetails";
+import NewMessageModal from "./NewMessageModal";
 
-function QueueTableRow(props: { row: QueueMessagesData }) {
+function QueueTableRow({
+  row,
+  fetchQueueMessages,
+  deleteMessage,
+  updateMessageExpandedState,
+}: {
+  row: QueueMessagesData;
+  fetchQueueMessages: (queueName: string) => Promise<void>;
+  deleteMessage: (
+    queueName: string,
+    messageId: string,
+    receiptHandle: string
+  ) => Promise<void>;
+  updateMessageExpandedState: (
+    queueName: string,
+    messageId: string | null
+  ) => void;
+}) {
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
 
-    const [isExpanded, setIsExpanded] = useState<boolean>(false);
-
-    function ExpandableArrowButton(props: { isExpanded: boolean }) {
-        return <IconButton aria-label="open-details" size="small"
-                           onClick={() => setIsExpanded(prevState => !prevState)}>
-            {props.isExpanded ? <KeyboardArrowRight/> : <KeyboardArrowDown/>}
-        </IconButton>
-    }
-
-    const {row} = props
+  function ExpandableArrowButton({ isExpanded }: { isExpanded: boolean }) {
```

Comment:
Hm... I don't think this needs to be declared in a function. `isExpanded` can be accessed directly from the state. Also, this is used only once here, so you could simply keep the returned jsx straight in the `TableCell` below

```tsx
<TableCell>
  <IconButton
    aria-label="open-details"
    size="small"
    onClick={() => setIsExpanded((prevState) => !prevState)}
  >
    {isExpanded ? <KeyboardArrowRight /> : <KeyboardArrowDown />}
  </IconButton>
</TableCell>
```